iroh-gateway/src/config.rs

```diff
@@ -0,0 +1,78 @@
+use std::collections::HashMap;
+
+use axum::http::header::*;
+
+pub static DEFAULT_PORT: u16 = 9050;
+
+#[derive(Debug, Clone)]
+pub struct Config {
+    pub writeable: bool,
+    pub fetch: bool,
+    pub cache: bool,
+    pub headers: HashMap<String, String>,
```

Comment:
https://docs.rs/headers/ is what `axum` uses under the hood, and will make header handling easier going forward (your call if you want to do this in a follow up or now)